inference-engine/src/vpu/graph_transformer/src/stages/loop_start.cpp

```diff
@@ -70,8 +71,53 @@ class LoopStart : public StageNode {
 
 }  // namespace
 
+namespace {
+
+class LoopDynamicIterNumber : public StageNode {
+public:
+    using StageNode::StageNode;
+
+protected:
+    StagePtr cloneImpl() const override {
+        return std::make_shared<LoopDynamicIterNumber>(*this);
+    }
+
+    // TODO: should this method be empty?
+    void propagateDataOrderImpl(StageDataInfo<DimsOrder>& orderInfo) override {
+    }
+
+    void getDataStridesRequirementsImpl(StageDataInfo<StridesRequirement>& stridesInfo) override {
+    }
+
+    void finalizeDataLayoutImpl() override {
+    }
+
+    void getBatchSupportInfoImpl(StageDataInfo<BatchSupport>& /*batchInfo*/) override {
+    }
+
+    void initialCheckImpl() const override {
+    }
+
+    void serializeParamsImpl(BlobSerializer& serializer) const override {
+        const auto numDims = inputEdge(0)->input()->desc().numDims();
+        const auto batchDimInd = numDims - 1 - dimToIeInd(Dim::N, numDims);
+        serializer.append(static_cast<uint32_t>(batchDimInd));
+    }
+
+    void serializeDataImpl(BlobSerializer& serializer) const override {
+        inputEdge(0)->input()->serializeBuffer(serializer);
+    }
+};
+
+}  // namespace
+
 Stage StageBuilder::addLoopStartStage(const Model& model, const std::string& name, const DataVector& inputs, const DataVector& outputs) {
     return model->addNewStage<LoopStart>(name, StageType::LoopStart, nullptr, inputs, outputs);
 }
 
+Stage StageBuilder::addLoopDynamicIterNumberStage(const Model& model, const std::string& name, const DataVector& inputs, const DataVector& outputs) {
+    return model->addNewStage<LoopDynamicIterNumber>(name, StageType::LoopDynamicIterNumber, nullptr, inputs, outputs);
+}
```

Comment:
I think stage expects exactly one input and output pair

Reply:
Fixed